applications/NXoptical_spectroscopy.nxdl.xml

```diff
@@ -0,0 +1,1121 @@
+                        counting (TCSPC).
+                    </doc>
+                </field>
+                <group name="device_information" type="NXfabrication" recommended="true"/>
+            </group>
+            <group name="source_TYPE" type="NXsource" nameType="partial" recommended="true">
+                <field name="type" recommended="true">
+                    <enumeration open="true">
+                        <item value="Synchrotron X-ray Source"/>
+                        <item value="Rotating Anode X-ray"/>
+                        <item value="Fixed Tube X-ray"/>
+                        <item value="UV Laser"/>
+                        <item value="Optical Laser"/>
+                        <item value="Laser"/>
+                        <item value="Dye-Laser"/>
+                        <item value="Broadband Tunable Light Source"/>
+                        <item value="Halogen lamp"/>
+                        <item value="LED"/>
+                        <item value="Mercury Cadmium Telluride"/>
+                        <item value="Deuterium Lamp"/>
+                        <item value="Xenon Lamp"/>
+                        <item value="Globar"/>
+                    </enumeration>
+                </field>
+                <field name="name" recommended="true"/>
+                <field name="standard" optional="true">
+                    <doc>
+                        If available, name/ID/norm of the light source standard.
+                    </doc>
+                </field>
+                <group name="device_information" type="NXfabrication" recommended="true">
+                    <doc>
+                        Details about the device information.
+                    </doc>
+                </group>
+                <field name="associated_beam" type="NX_CHAR" recommended="true">
+                    <doc>
+                        The path to a beam emitted by this source.
+                        Should be named with the same appendix, e.g.,
+                        for TYPE=532nmlaser, there should as well be
+                        a NXbeam named "beam_532nmlaser" together with this source
+                        instance named "source_532nmlaser"
+                        
+                        Example: /entry/instrument/beam_532nmlaser
+                    </doc>
+                </field>
+            </group>
+            <group type="NXmonochromator" recommended="true">
+                <group name="device_information" type="NXfabrication" recommended="true"/>
+            </group>
+            <field name="angle_reference_frame" recommended="true">
+                <doc>
+                    Defines the reference frame which is used to describe the sample orientation
+                    with respect to the beam directions.
+                    
+                    A beam centered description is the default and uses 4 angles(similar to XRD):
```

Comment:
Is there a better way to describe both situations using a CS and transformations for a beam and a sample stage?

Reply:
I checked if the implementation in the classes and appdefs that was voted upon resolved this comment - it does.

Reply:
The last comment in the docstring though points nowhere, as a concept references_frames does not exist in NXoptical_spectroscopy

```An arbitrary reference frame can be defined by "reference_frames"
   and used via "instrument/angle_sample_and_beam_TYPE"```